src/main.cpp

```diff
@@ -147,51 +147,77 @@ int main(int argc, char** argv)
     auto applyCliArgs = [&](Settings* settings) {
         if (parser.isSet(sysroot)) {
             settings->setSysroot(parser.value(sysroot));
+            settings->setLastUsedEnvironment({});
```

Comment:
why? also, please simplify this repetitive code via lambdas:

```
auto applyCliArg = [&](const QCommandLineOption &arg, auto setter) {
    if (parser.isSet(arg)) {
        (settings->*setter)(parser.value(arg));
        // comment to explain why this is done
        settings->setLastUsedEnvironment({});
    }
};
applyCliArg(kallsyms, &Settings::setKallsyms);
...
```

Reply:
otherwise its broken.

in mainwindow 296 we have
```c++
m_lastUsedSettings = m_config->group("PerfPaths").readEntry("lastUsed");
if (!m_lastUsedSettings.isEmpty()) {
    auto currentConfig = m_config->group("PerfPaths").group(m_lastUsedSettings);
    settings->setSysroot(currentConfig.readEntry("sysroot", ""));
    settings->setAppPath(currentConfig.readEntry("appPath", ""));
    settings->setExtraLibPaths(currentConfig.readEntry("extraLibPaths", ""));
    settings->setDebugPaths(currentConfig.readEntry("debugPaths", ""));
    settings->setKallsyms(currentConfig.readEntry("kallsyms", ""));
    settings->setArch(currentConfig.readEntry("arch", ""));
    settings->setObjdump(currentConfig.readEntry("objdump", ""));
}
```
which will overwrite the sysroot, ... if lastUsed is not empty